server/data/incidentClient.ts

```diff
@@ -280,4 +282,36 @@ export default class IncidentClient {
       values: [formResponse, incidentDate, reportId],
     })
   }
+
+  async getReportInProgress(bookingId: number, status: string): Promise<inProgressReport> {
+    const result = await this.query({
+      text: `
+      select vr.id id, vr.incident_date as incidentdate
+      from v_report vr 
+      where booking_id = $1
+      and status = $2`,
+      values: [bookingId, status],
+    })
+    return result.rows[0]
+  }
+
+  async getReportsForAnOffenderForSpecificDate(
+    bookingId: number,
+    [startDate, endDate]: DateRange
+  ): Promise<offenderReports[]> {
+    const results = await this.query({
+      text: `
+      select vr.incident_date date, vr.form_response form, vr.reporter_name reporter, vr.status status
+      from v_report vr where vr.offender_no 
+      = (
+        select vr2.offender_no from v_report vr2 
+        where vr2.booking_id = $1
+        limit 1
+        )
+      and vr.incident_date >= $2
+      and vr.incident_date < $3`,
+      values: [bookingId, startDate, endDate],
```

Comment:
You don't need the sub select as v_report already has the booking ID

Reply:
done